source/server/http/admin.cc

```diff
@@ -1147,5 +1147,19 @@ Http::Code AdminImpl::request(absl::string_view path_and_query, absl::string_vie
   return code;
 }
 
+void AdminImpl::closeSocket() {
+  if (socket_) {
+    socket_->close();
+  }
+}
+
+void AdminImpl::addListenerToHandler(Network::ConnectionHandler* handler) {
+  if (listener_) {
+    handler->addListener(*listener_);
+  } else {
+    throw EnvoyException("asked to add a non-existent listener");
```

Comment:
Is this added just for the test? I wonder if there is some other way we can test this without adding logic here that we don't otherwise need? Thoughts?